File: 169-majority-element/solution.cpp
#include <vector>

class Solution
{
public:
  int majorityElement(std::vector<int> &nums)
  {
    int val = 0;
    int val_occur = 0;
    for (const int num : nums)
    {
      if (val_occur <= 0)
      {
        val = num;
        val_occur = 1;
        continue;
      }

      if (val == num)
      {
        ++val_occur;
        continue;
      }

      --val_occur;
    }

    return val;
  }
};

File: 169-majority-element/solution.py
from typing import List


class Solution:
  def majorityElement(self, nums: List[int]) -> int:
    val = 0
    val_occur = 0
    for num in nums:
      if val_occur <= 0:
        val = num
        val_occur = 1
        continue

      # val_occur > 0
      if val == num:
        val_occur += 1
        continue

      # val != num
      val_occur -= 1

    return val
